src/graph/entities/edge.h

```diff
@@ -14,6 +14,9 @@ typedef int RelationID;
 
 #define EDGE_LENGTH_INF UINT_MAX - 2
 
+// defined in node.h
+extern AttributeSet NULL_ATTRIBUTE_SET;
+
```

Comment:
see comment on `node.h` I think `AttributeSet NULL_ATTRIBUTE_SET` should be defined in: `src/graph/entities/attribute_set.h`